src/serialization/mod.rs

```diff
@@ -80,23 +80,43 @@ impl<T: CipherSuite> From<&LoginThirdMessage<T>> for ProtocolMessageType {
     }
 }
 
-pub(crate) fn serialize(input: &[u8], max_bytes: usize) -> Vec<u8> {
-    let mut output: Vec<u8> = Vec::new();
-    output.extend_from_slice(&input.len().to_be_bytes()[8 - max_bytes..]);
-    output.extend_from_slice(&input[..]);
+// Corresponds to the I2OSP() function from RFC8017
+pub(crate) fn i2osp(input: usize, length: usize) -> Vec<u8> {
+    if length <= 8 {
+        return (&input.to_be_bytes()[8 - length..]).to_vec();
+    }
+
+    let mut output = vec![0u8; length];
+    output.splice(length - 8..length, input.to_be_bytes().iter().cloned());
     output
 }
 
+// Corresponds to the OS2IP() function from RFC8017
+pub(crate) fn os2ip(input: &[u8]) -> Result<usize, PakeError> {
+    if input.len() > 8 {
+        return Err(PakeError::SerializationError);
+    }
+
+    let mut output_array = [0u8; 8];
+    for i in 0..input.len() {
+        output_array[8 - input.len() + i] = input[i];
```

Comment:
Would it be simpler to use `copy_from_slice`, considering it takes a `&mut self`?